python_modules/libraries/dagster-components/dagster_components/lib/dbt_project/component.py

```diff
@@ -104,7 +110,20 @@ def get_asset_selection(
     def build_defs(self, context: ComponentLoadContext) -> Definitions:
         self.project.prepare_if_dev()
 
-        @dbt_assets(
+        config_arg = inspect.signature(self.execute).parameters.get("config")
+
+        def _run_dbt_assets_fn(context: AssetExecutionContext, config: Config):
+            # pass config to the execute method if it exists on the signature
+            if config_arg:
+                yield from self.execute(context=context, dbt=self.dbt, config=config)  # type: ignore
```

Comment:
we should change the signature of `execute` to have kwargs rather than do the type ignore

Reply:
Good call, updated